File: Golang/636. Exclusive Time of Functions.go
func exclusiveTime(n int, logs []string) []int {
    stack := make([][]int, 0, n)
    ans := make([]int, n)
    for _, log := range logs {
        item := strings.Split(log, ":")
        uid, _ := strconv.Atoi(item[0])
        time, _ := strconv.Atoi(item[2])
        
        if item[1] == "start" {
            stack = append(stack, []int{time, 0})
        } else {
            output := stack[len(stack)-1]
            stack = stack[: len(stack)-1]
            ans[uid] += time - output[0] + 1 - output[1]
            if len(stack) > 0 {
                stack[len(stack)-1][1] += time - output[0] + 1
            }
        }
    }
    return ans
}

File: Python3/636. Exclusive Time of Functions.py
class Solution:
    def exclusiveTime(self, n, logs):
        """
        :type n: int
        :type logs: List[str]
        :rtype: List[int]
        """
        ans = [0 for _ in range(n)]
        stack = []
        for log in logs:
            uid, status, time = map(int, log.replace('start', '1').replace('end', '0').split(":"))
            if status:
                stack.append([time, 0])
            else:
                starttime, timecost = stack.pop()
                ans[uid] += time - starttime + 1 - timecost
                if stack:
                    stack[-1][1] += time - starttime + 1
        return ans
